utility_network/perform-valve-isolation-trace/src/main/java/com/esri/samples/perform_valve_isolation_trace/PerformValveIsolationTraceController.java

```diff
@@ -0,0 +1,308 @@
+      FeatureLayer distributionLineLayer = new FeatureLayer(distributionLineFeatureTable);
+
+      ServiceFeatureTable deviceFeatureTable = new ServiceFeatureTable(featureServiceURL + "/0");
+      FeatureLayer deviceLayer = new FeatureLayer(deviceFeatureTable);
+
+      // add the feature layers to the map
+      map.getOperationalLayers().addAll(Arrays.asList(distributionLineLayer, deviceLayer));
+
+      // create a graphics overlay for the starting location and add it to the map view
+      GraphicsOverlay startingLocationGraphicsOverlay = new GraphicsOverlay();
+      mapView.getGraphicsOverlays().add(startingLocationGraphicsOverlay);
+
+      // create and apply renderers for the starting point graphics overlay
+      SimpleMarkerSymbol startingPointSymbol =
+              new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CROSS, ColorUtil.colorToArgb(Color.LIGHTGREEN), 25);
+      startingLocationGraphicsOverlay.setRenderer(new SimpleRenderer(startingPointSymbol));
+
+      // create and load the utility network
+      utilityNetwork = new UtilityNetwork(featureServiceURL, map);
+      utilityNetwork.loadAsync();
+      utilityNetwork.addDoneLoadingListener(() -> {
+        if (utilityNetwork.getLoadStatus() == LoadStatus.LOADED) {
+
+          // get a trace configuration from a tier
+          UtilityNetworkDefinition networkDefinition = utilityNetwork.getDefinition();
+          UtilityDomainNetwork domainNetwork = networkDefinition.getDomainNetwork("Pipeline");
+          UtilityTier tier = domainNetwork.getTier("Pipe Distribution System");
+          traceConfiguration = tier.getTraceConfiguration();
+
+          // create a trace filter
+          traceConfiguration.setFilter(new UtilityTraceFilter());
+
+          // get a default starting location
+          UtilityNetworkSource networkSource = networkDefinition.getNetworkSource("Gas Device");
+          UtilityAssetGroup assetGroup = networkSource.getAssetGroup("Meter");
+          UtilityAssetType assetType = assetGroup.getAssetType("Customer");
+          startingLocation = utilityNetwork.createElement(assetType, UUID.fromString("98A06E95-70BE-43E7-91B7-E34C9D3CB9FF"));
+
+          // get the first feature for the starting location, and get its geometry
+          ListenableFuture<List<ArcGISFeature>> elementFeaturesFuture =
+                  utilityNetwork.fetchFeaturesForElementsAsync(Collections.singletonList(startingLocation));
+
+          elementFeaturesFuture.addDoneListener(() -> {
+            try {
+              List<ArcGISFeature> startingLocationFeatures = elementFeaturesFuture.get();
+
+              if (!startingLocationFeatures.isEmpty()) {
+                Geometry startingLocationGeometry = startingLocationFeatures.get(0).getGeometry();
+
+                if (startingLocationGeometry instanceof Point){
+                  Point startingLocationGeometryPoint = (Point) startingLocationFeatures.get(0).getGeometry();
```

Comment:
to avoid repeating the code in line 141, looks like we could use: 
```suggestion
                  Point startingLocationGeometryPoint = (Point) startingLocationGeometry;
```